src/sagemaker/workflow/_utils.py

```diff
@@ -291,26 +295,26 @@ class _RegisterModelStep(ConfigurableRetryStep):
     def __init__(
         self,
         name: str,
-        content_types,
-        response_types,
-        inference_instances,
-        transform_instances,
-        estimator: EstimatorBase = None,
-        model_data=None,
-        model_package_group_name=None,
-        model_metrics=None,
-        metadata_properties=None,
-        approval_status="PendingManualApproval",
-        image_uri=None,
-        compile_model_family=None,
-        display_name: str = None,
-        description=None,
-        depends_on: Union[List[str], List[Step]] = None,
-        retry_policies: List[RetryPolicy] = None,
-        tags=None,
-        container_def_list=None,
-        drift_check_baselines=None,
-        customer_metadata_properties=None,
+        content_types: List,
+        response_types: List,
+        inference_instances: List,
+        transform_instances: List,
+        estimator: Optional[EstimatorBase] = None,
+        model_data: Optional[str] = None,
```

Comment:
This could be a pipeline variable. I wonder if we can annotate this, which makes documentation and test easier.

Reply:
yea, but theoretically, as long as the arg is primitive type (even a nested arg for a python instance), it can be parameterized right? unless we go through how it is used top down in the codebase, we can't just make the empirical pick, right?